src/gravel/controllers/bootstrap.py

```diff
@@ -77,6 +77,15 @@ async def bootstrap(self) -> bool:
 
         return True
 
+    async def mark_finished(self) -> bool:
+        stage: DeploymentStage = gstate.config.deployment_state.stage
+        if stage == DeploymentStage.ready:
+            return True
+        elif stage != DeploymentStage.bootstrapped:
+            return False
```

Comment:
it feels as if `DeploymentStage.ready` really is the same as `DeploymentStage.bootstrapped` ...

Reply:
it isn't though - bootstrapped means we can now start deploying services, whereas 'ready' means we have gone through the whole installer.